projects/plugins/boost/app/modules/optimizations/page-cache/pre-wordpress/class-filesystem-utils.php

```diff
@@ -198,10 +201,6 @@ public static function gc_expired_files( $directory, $file_ttl, $action = self::
 		}
 		closedir( $handle );
 
-		if ( $action === self::REBUILD ) {
-			return $count;
-		}
-
 		// If the directory is empty after processing its files, delete it.
```

Comment:
Seems like we are handling index.html in a special way and only deleting it when deleting the directory. Check the definition of `is_dir_empty`:

https://github.com/Automattic/jetpack/blob/dc83c8e07bc72912269718224c70ace6f8bfd2e9/projects/plugins/boost/app/modules/optimizations/page-cache/pre-wordpress/class-filesystem-utils.php#L314C2-L321C3

For some reason that approach is not working. I think we should fix that instead.